parsec/api/protocol/base.py

```diff
@@ -128,10 +128,19 @@ class TimestampOutOfBallparkRepSchema(BaseRepSchema):
     # `bad_timestamp` is kept for backward compatibility,
     # even though `timestamp_out_of_ballpark` would be more explicit
     status: fields.CheckedConstant = fields.CheckedConstant("bad_timestamp", required=True)
-    ballpark_client_early_offset = fields.Float(required=False, missing=None)
-    ballpark_client_late_offset = fields.Float(required=False, missing=None)
-    client_timestamp = fields.DateTime(required=False, missing=None)
-    backend_timestamp = fields.DateTime(required=False, missing=None)
+    ballpark_client_early_offset = fields.Float(required=False, allow_none=False)
+    ballpark_client_late_offset = fields.Float(required=False, allow_none=False)
+    client_timestamp = fields.DateTime(required=False, allow_none=False)
+    backend_timestamp = fields.DateTime(required=False, allow_none=False)
+
+    @post_load
+    def make_obj(self, data: Dict[str, Any]) -> Dict[str, Any]:  # type: ignore[misc]
+        # Cannot use `missing=None` with `allow_none=False`
+        data.setdefault("ballpark_client_early_offset", None)
+        data.setdefault("ballpark_client_late_offset", None)
+        data.setdefault("client_timestamp", None)
+        data.setdefault("backend_timestamp", None)
+        return data
```

Comment:
If the logic doesn't change, shouldn't we keep the original version as it seems more explicit?

Reply:
My commit comment was wrong: the original code uses `allow_none=True`, cf. what is done in marshmallow:

```python
        # If missing=None, None should be considered valid by default
        if allow_none is None:
            if missing is None:
                self.allow_none = True
            else:
                self.allow_none = False
        else:
            self.allow_none = allow_none
```

This is typically for this kind of "too smart for it own good" code that we need to be explicit about required/allow_none !

I'm changing the commit comment then